git-link.el

```diff
@@ -142,10 +142,22 @@
   (save-restriction
     (widen)
     (save-excursion
-      (let ((use-region (use-region-p)))
-        (list
-         (line-number-at-pos (when use-region (region-beginning)))
-         (when use-region (line-number-at-pos (region-end))))))))
+      (let* ((use-region (use-region-p))
+             (start (when use-region (region-beginning)))
+             (end   (when use-region (region-end)))
+             (line-start (line-number-at-pos start))
+             line-end)
+        (when use-region
+          ;; If the `end' position is at the beginning of a line
+          ;; decrement the position by 1, so that the resultant
+          ;; position is on the previous line
+          (when (= end (line-beginning-position))
```

Comment:
This does not account for the reverse scenario: the point position is less than the mark position. In this case the extraneous line is still included. For example, `*P*` is point, `*M*` is mark:

```
1. *P*one
2. two
3. three
4. *M*
```
calling `git-link` with the arraignment will include lines 1-4.  

Need to check if `(region-end)` is at the beginning of the line, not `(point)`. Off hand not sure how to do this without moving point. Ideas?